### DISCUSSION TITLE
Ello! I like to ask if it's possible to use the rldev decompiled seen.utf (the extracted text) with SExtractor? Also have a new key for seen_fix.py!

### DISCUSSION THREAD

**Discussion by Linden10** [satan53x/SExtractor#124] [Q&A] [answered]
Ello, it's me Linden10 from before when I asked for help on reallive!
Been translating the TANE game very well and recently gotten the other games by CONCEPT as well!

I had a fellow super helpful user on discord helped me with decrypting the game [HARAKAMI ](https://vndb.org/v6022) which had alpharom, region locking and encryption on the seen.txt! He helped guide me on how to remove the region locking and extract the decryption key!

He was able to test and successfully extract the SEEN.txt file with rldev and even gave me a the decompiled .utf/.org seen files for translation, which can allow "!" and "" in the text along with /n for linebreak thanks to rldev.

I was wondering if it's possible to use the .utf files with SExtractor? I'll provide the full files that the dev shared here!
https://pixeldrain.com/u/evD4LpRU

And also I like to provide the decryption key for HARAKAMI which you can add to seen_fix.py! Here you go:

`67 17 21 BE 6D 27 B5 21 4A 82 39 0E AD C6 73 EE`

Here's the seen_fix.py format too:
```
	#Harakami (孕神～はらかみ～)
	69: bytearray.fromhex('67 17 21 BE 6D 27 B5 21 4A 82 39 0E AD C6 73 EE'),
```

Edit: I feel the need to give credits to the user! ^^
 I gave thanks to Adrianrhod of [Eroge Translation Center](https://discord.gg/TNFVTGEU74) for helping me out thus far!

**Comment by satan53x** [comment, marked as answer]
Of course they can be used together. This is a secondary extraction solution. 
You can choose the TXT engine and write some simple regular expressions for processing, such as:
```
10_search=^<\d+> \\{(?P<name>.+?)}(.+)$
15_search=^<\d+> \\?(.+)$
```

However, rldev is more dependent on the game version, you can refer to here: 
https://github.com/satan53x/SExtractor/issues/3#issuecomment-2109283750

**Reply by Linden10** [reply]
Hi, so I successfully extracted and translated one of the .utf files with DazedMTL and than import them back into an .utf file!

Now it seems everything is almost working but there's currently an tiny issue that's preventing them from being converted into a seen.txt file! Very tiny!
That issue being...that some of the lines aren't properly spaced when they don't have a name tag, thus the "\" next to the letters gets treated like some variable which rldev errors out on like so:
```
<0000> \{Teacher}“Alright, let's start the first homeroom of the second\nyear.”
<0001> \As the homeroom teacher for this year made that announcement\nfrom the podium, the classroom returned to silence.
<0002> \Just a moment ago, lively voices saying things like "nice to\nmeet you this year" and "pleased to meet you," were flying\naround.
<0003> \{Susumu}“.........”
<0004> \I was absent-mindedly watching this scene from a seat a\nlittle towards the back in the middle of the classroom.
<0005> \About a third of the classmates were in the same class as me\nlast year.
<0006> \The rest are new classmates this year.
<0007> \Honestly, I don't really care much about who makes up the\nclass.
<0008> \{Teacher}“Ah, first about the seating... for that, we'll draw lots\nunder the supervision of the day duty tomorrow morning.”
<0009> \{Teacher}“Right now, let's prioritize the announcements.”
<0010> \The seats we're sitting in now were chosen arbitrarily after\nthe entrance ceremony.
```

It should look something like this:
```
<0000> \{教師}「よぉし、二年最初のホームルームを始めるぞ」
<0001> \　教壇に立った今年の担任がそう宣言すると、教室内は静けさを取り戻した。
<0002> \　ついさっきまで、今年もよろしくだの、初めましてだの、賑やかな声が飛び交っていたにもかかわらず、だ。
<0003> \{勧}「…………」
<0004> \　俺は、そんな様子を教室中程より、少し後ろの席でぼーっと眺めていた。
<0005> \　クラスメートの三分の一程度は、一年の時に同じクラスだった連中。
<0006> \　残りは、今年新しくクラスメートとなった連中だ。
<0007> \　正直、クラスのメンツなんてのは、割とどうでもいいのだが。
<0008> \{教師}「あー、まず座る席だが……それについては、明日の朝にでも日直の監督の下にくじ引きで決めてくれ」
<0009> \{教師}「今は連絡事項を優先するからな」
<0010> \　今座っている席は、各自、始業式の後に適当に選んだものだ。
```

Very small issue, should I adjust this regex to account for that space?
15_search=^<\d+> \\?(.+)$

Feel like this is an super easy solution...
Once I successfully get the translation in game, I'll mark your answer and close the thread heh.

**Reply by Cosetto** [reply]
The space is probably line indent, so it's up to you to remain or delete it

**Reply by Linden10** [reply]
> The space is probably line indent, so it's up to you to remain or delete it

Ello! The space is indeed a space as I confirmed it in notepad++, it's required otherwise rldev will throw errors like this:
```
Compiling SEEN1030.TXT...
  Error (SEEN1030.utf line 10): expected expression, found undeclared
  identifier[8] `A'.
```

So all I need is a way to make the space remain when extracting the sentence...yup!

**Reply by Linden10** [reply]
AH I just figured out the issue! It seems that the lines when translated by DazedMTL loses the space!
I'll look into fixing that, sorry to bother and thanks for helping!
I'll just mark your answer and close the thread heh.

**Reply by satan53x** [reply]
You can move the space outside the capturing group `(.+)`:
```
10_search=^<\d+> \\{(?P<name>.+?)}(.+)$
15_search=^<\d+> \\?　?(.+)$
```
Or split into two searches:
```
10_search=^<\d+> \\{(?P<name>.+?)}(.+)$
15_search=^<\d+> \\　(.+)$
16_search=^<\d+> (.+)$
```
Both of the above methods will retain spaces and not be extracted, so they will not be translated.

**Reply by Linden10** [reply]
Thanks for helping! I’ll test these out! 😊

**Reply by Linden10** [reply]
Okay! So I had to change the space from `"\\?　?"` in your first suggested regex a bit to this `"\\?　?"` and than it worked!
I copied the space/line indent from the xlsx files that had the space in MS Excel into the regex and this worked:
`15_search=^<\d+> \\?　?(.+)$`

Now the lines are properly extracted and the translations are added back in without issue!
Thank you for helping heh! Now I don't have to worry about DazedMTL not adding the space at the beginning of the line! ^_^

**Comment by Cosetto** [comment]
Actually, for Hadashi Shoujo and other brands, you can choose either 1.5.9.1 or 1.6.0.1, both will work.
![image](https://github.com/user-attachments/assets/9651b675-826a-4e46-bd2b-727addb6850b)
But mostly, you can view the version by pointing mouse to Reallive.exe or check it in properties.

**Comment by Linden10** [comment]
> Of course they can be used together. This is a secondary extraction solution. You can choose the TXT engine and write some simple regular expressions for processing, such as:
> 
> ```
> 10_search=^<\d+> \\{(?P<name>.+?)}(.+)$
> 15_search=^<\d+> \\?(.+)$
> ```
> 
> However, rldev is more dependent on the game version, you can refer to here: [#3 (comment)](https://github.com/satan53x/SExtractor/issues/3#issuecomment-2109283750)

Ah thanks for helping, I'll test it out with the files heh!
So those regular expressions will be able to detect the .utf files eh?
~Or should I separate the .utf files from the .org files? Just wondering! ^^~

Edit: Actually I just tested SExtractor with the regex you shared and it successfully extracted
only the .utf files which is great! Seems the lines are properly separated too, will test
if it successfully translates the lines.

Here's an example of the extracted lines in SExtractor (taken from an xlsx file which is why it looks like this):
```
SEEN0123.utf.xlsx
Key | Value
-- | --
冥那 |  
「はい。こちらこそよろしくおねがいします」 |  
「自分を表す数字ですか？　そうですね…１７でしょうか」 |  
「河瀬美冥那です。あなたと出会い、いっしょに赤ちゃんをつくるのを、楽しみにしています」 |  
翡翠 |  
「痛みも刺激。私にあなたを刻みつけて欲しい…」 |  
「私を表す数字。強いてあげるなら１９」 |  
「翡翠。死神。あなたは私で興奮し、受胎させるべき」 |  
リューズ |  
「ニンゲン、そのようないやらしい目で見るな！」 |  
「リューズを表す数字？　１０…だな」 |  
「リューズ。ツクモ神だ。ニンゲンにしては見所があるな。特別にリューズと子作りさせてやろう」 |  
恵 |  
「ケイ、それくらい知ってる！　あっ、当たり前じゃない！」 |  
「ケイを表す数字か～。１８と言いたいけど、実際は１２ぐらい？」 |  
「リンドウケイだよ。ケイの子宮で精子と卵子を受精させるんでしょ？　ケイ、オトナだから大丈夫だよ！」 |  
狐呱 |  
「この一帯を治める土地神じゃ。フフフ。あがめるが良いぞ」 |  
「わらわを表す数字とな？　フム…１４と言った所か」 |  
「土地神の狐呱じゃ。みごとわらわに中出しして身篭もらせてみよ」 |  
ようこ |  
「ようこ。ひらがなでようこって書くんだ。よろしくな！」 |  
「ボクを表す数字？　何だろ……１６、かな」 |  
「もがみようこだよ。早くボクを妊娠させに来てよ。楽しみにまってるから！」 |  

```

Hmm...rldev is dependent on the game version and it may not work as well with SExtractor based on the screenshot...
Interesting to know! But I see [Cosetto](https://github.com/Cosetto) has already figured it out!

> Actually, for Hadashi Shoujo and other brands, you can choose either 1.5.9.1 or 1.6.0.1, both will work. ![image](https://private-user-images.githubusercontent.com/81801486/397380520-9651b675-826a-4e46-bd2b-727addb6850b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2Mzg1MzAsIm5iZiI6MTczNDYzODIzMCwicGF0aCI6Ii84MTgwMTQ4Ni8zOTczODA1MjAtOTY1MWI2NzUtODI2YS00ZTQ2LWJkMmItNzI3YWRkYjY4NTBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE5NTcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMTExOGU0ZjA2ZDNiZjFjMWQzYzdhNDM2ZTQ1ZGY5OWRhMTExYWNlNjlhYjU5ODVjODAxYTY4NGM3YWQ0MzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8f1kvJZ2fpJBRSuNPHPVse42CsAiai1n6Yp6zn_XsfM) But mostly, you can view the version by pointing mouse to Reallive.exe or check it in properties.

I checked the reallive version that HARAKAMI uses and it uses 1.6.5.2 for reallive.
As satan53x said, does the versions imply how rldev will work or no with SExtractor?

The game.cfg you showed...AH I found it, you can found it in the pixeldrian link I shared.
How very kind of them helping me out this much.

```
game HA "Harakami"
	for RealLive 1.6.0.1
	
	using key from 256 for 257:
		0x67 0x17 0x21 0xBE 0x6D 0x27 0xB5 0x21
		0x4A 0x82 0x39 0x0E 0xAD 0xC6 0x73 0xEE
```

At this point, I feel a dis-service not crediting them. Sorry sorry.
I gave thanks to Adrianrhod of [Eroge Translation Center](https://discord.gg/TNFVTGEU74) for helping me out thus far!

Anywho, the .utf scripts will be compiled in rl 1.6.0.1 so I guess this'll hopefully work!
I guess changing the version up will not make it work as well?
